frontend/packages/console-app/src/components/volume-snapshot/volume-snapshot.tsx

```diff
@@ -37,189 +37,273 @@ import {
-    <>
-      <TableData {...tableColumnInfo[0]}>
-        <ResourceLink
-          kind={referenceForModel(VolumeSnapshotModel)}
-          name={name}
-          namespace={namespace}
-        />
-      </TableData>
-      <TableData {...tableColumnInfo[1]} columnID="namespace">
-        <ResourceLink kind={NamespaceModel.kind} name={namespace} />
-      </TableData>
-      <TableData {...tableColumnInfo[2]}>
-        <Status status={volumeSnapshotStatus(obj)} />
-      </TableData>
-      <TableData {...tableColumnInfo[3]}>{sizeMetrics}</TableData>
-      {!customData?.disableItems?.Source && (
-        <TableData {...tableColumnInfo[4]}>
+    const rowCells = {
+      [tableColumnInfo[0].id]: {
+        cell: <ResourceLink kind={kind} name={name} namespace={namespace} />,
+        props: getNameCellProps(name),
+      },
+      [tableColumnInfo[1].id]: {
+        cell: <ResourceLink kind={NamespaceModel.kind} name={namespace} />,
+      },
+      [tableColumnInfo[2].id]: {
+        cell: <Status status={volumeSnapshotStatus(obj)} />,
+      },
+      [tableColumnInfo[3].id]: {
+        cell: sizeMetrics,
+      },
+      [tableColumnInfo[4].id]: {
+        cell: (
           <ResourceLink
             kind={referenceForModel(sourceModel)}
             name={sourceName}
             namespace={namespace}
           />
-        </TableData>
-      )}
-      {!customData?.disableItems?.['Snapshot Content'] && (
-        <TableData {...tableColumnInfo[5]}>
-          {snapshotContent ? (
-            <ResourceLink
-              kind={referenceForModel(VolumeSnapshotContentModel)}
-              name={snapshotContent}
-            />
-          ) : (
-            '-'
-          )}
-        </TableData>
-      )}
-      <TableData {...tableColumnInfo[6]}>
-        {snapshotClass ? (
+        ),
+      },
+      [tableColumnInfo[5].id]: {
+        cell: snapshotContent ? (
+          <ResourceLink
+            kind={referenceForModel(VolumeSnapshotContentModel)}
+            name={snapshotContent}
+          />
+        ) : (
+          DASH
+        ),
+        disabled: hideSnapshotContentColumn,
+      },
+      [tableColumnInfo[6].id]: {
+        cell: snapshotClass ? (
           <ResourceLink kind={referenceForModel(VolumeSnapshotClassModel)} name={snapshotClass} />
         ) : (
-          '-'
-        )}
-      </TableData>
-      <TableData {...tableColumnInfo[7]}>
-        <Timestamp timestamp={creationTimestamp} />
-      </TableData>
-      <TableData className={Kebab.columnClass}>
-        <LazyActionMenu context={context} />
-      </TableData>
-    </>
+          DASH
+        ),
+      },
+      [tableColumnInfo[7].id]: {
+        cell: <Timestamp timestamp={creationTimestamp} />,
+      },
+      [tableColumnInfo[8].id]: {
+        cell: <LazyActionMenu context={context} />,
+        props: {
+          ...actionsCellProps,
+        },
+      },
+    };
+
+    return columns
+      .filter(({ id }) => !rowCells[id].disabled)
```

Comment:
`disabled` isn't always defined in each rowCells, is `disabled` safe to access without chaining?